receiver/googlecloudmonitoringreceiver/internal/metrics_conversion.go

```diff
@@ -226,10 +226,34 @@ func (mb *MetricsBuilder) ConvertDistributionToMetrics(ts *monitoringpb.TimeSeri
 			continue
 		}
 
-		countTotal := uint64(0)
 		targetBucketCounts := targetDataPoint.BucketCounts()
-		targetBucketCounts.EnsureCapacity(len(sourceBucketCounts))
-		for _, bucketCount := range sourceBucketCounts {
+
+		// The number of elements in bucket_counts array must be by one greater than the number of elements in explicit_bounds array.
+		lenBounds := targetDataPoint.ExplicitBounds().Len()
+		lenCounts := lenBounds + 1
+
+		// The exception to this rule is when the length of bucket_counts is 0, then the length of explicit_bounds must also be 0.
+		// Google says this case should never happen, but you never know...
```

Comment:
To give you a bit more confidence, Google's monitoring API currently rejects points with zero-length bucket counts.